include/deal.II/numerics/vector_tools.h

```diff
@@ -698,26 +698,26 @@ namespace VectorTools
    *
    * See the general documentation of this namespace for more information.
    */
-  template <typename DoFHandlerType>
+  template <typename number, typename DoFHandlerType>
```

Comment:
here and below: you make an incompatible change by adding the new template argument as first argument. This breaks all code where someone already specifies the dof handler type explicitly. better to add the new template argument last.